starport/cmd/network_chain_publish.go

```diff
@@ -47,16 +53,27 @@ func NewNetworkChainPublish() *cobra.Command {
 
 func networkChainPublishHandler(cmd *cobra.Command, args []string) error {
 	var (
-		source        = args[0]
-		tag, _        = cmd.Flags().GetString(flagTag)
-		branch, _     = cmd.Flags().GetString(flagBranch)
-		hash, _       = cmd.Flags().GetString(flagHash)
-		genesisURL, _ = cmd.Flags().GetString(flagGenesis)
-		chainID, _    = cmd.Flags().GetString(flagChainID)
-		campaign, _   = cmd.Flags().GetUint64(flagCampaign)
-		noCheck, _    = cmd.Flags().GetBool(flagNoCheck)
+		source            = args[0]
+		tag, _            = cmd.Flags().GetString(flagTag)
+		branch, _         = cmd.Flags().GetString(flagBranch)
+		hash, _           = cmd.Flags().GetString(flagHash)
+		genesisURL, _     = cmd.Flags().GetString(flagGenesis)
+		chainID, _        = cmd.Flags().GetString(flagChainID)
+		campaign, _       = cmd.Flags().GetUint64(flagCampaign)
+		noCheck, _        = cmd.Flags().GetBool(flagNoCheck)
+		mainnet, _        = cmd.Flags().GetBool(flagMainnet)
+		totalSupplyStr, _ = cmd.Flags().GetString(flagTotalSupply)
 	)
 
+	if mainnet && campaign == 0 && totalSupplyStr == "" {
+		return errors.New("you should use the --mainnet flag along with the --campaign or --total-supply flag")
```

Comment:
```suggestion
		return errors.New("%s flag requires on of the %s or %s flags to be set", flagX...)
```

flag names are already assigned to constants, we can instead use them.